.github/workflows/test-all.yaml

```diff
@@ -37,13 +37,73 @@ jobs:
             "rust",
             "sshd",
             "terraform",
+        ]
+        baseImage:
+          [
+            "ubuntu:focal",
+            "ubuntu:jammy",
+            "debian:11",
+            "mcr.microsoft.com/vscode/devcontainers/base:ubuntu",
+            "mcr.microsoft.com/devcontainers/base:debian",
           ]
-        baseImage: [ "ubuntu:focal", "ubuntu:jammy", "debian:11", "mcr.microsoft.com/vscode/devcontainers/base:ubuntu", "mcr.microsoft.com/vscode/devcontainers/base:debian" ]
     steps:
       - uses: actions/checkout@v2
 
       - name: "Install latest devcontainer CLI"
         run: npm install -g @devcontainers/cli
 
-      - name: " Testing '${{ matrix.features }}' against '${{ matrix.baseImage }}'"
-        run: devcontainer features test -f ${{ matrix.features }} -i ${{ matrix.baseImage }} -c `pwd`
+      - name: "Generating tests for '${{ matrix.features }}' against '${{ matrix.baseImage }}'"
+        run: devcontainer features test  --skip-scenarios -f ${{ matrix.features }} -i ${{ matrix.baseImage }} .
+
+  test-scenarios:
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    strategy:
+      matrix:
+        features: [
+            "anaconda",
+            "aws-cli",
+            "azure-cli",
+            "common-utils",
+            "desktop-lite",
+            "docker-from-docker",
+            "docker-in-docker",
+            "dotnet",
+            "git",
+            "git-lfs",
+            "github-cli",
+            "go",
+            "hugo",
+            "java",
+            "kubectl-helm-minikube",
+            "node",
+            "nvidia-cuda",
+            "oryx",
+            "php",
+            "powershell",
+            "python",
+            "ruby",
+            "rust",
+            "sshd",
+            "terraform",
+        ]
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: "Install latest devcontainer CLI"
+        run: npm install -g @devcontainers/cli
```

Comment:
Wondering if we could install `@devcontainers/cli` only once at some top level which can then be consumed by all jobs.

Reply:
These run on different vms in parallel. Not sure if there is a "beforeAll().." hook for actions.